hw2/01-coins.js

```diff
@@ -1,7 +1,33 @@
 /** Exercise 01 - Coins **/
 
 const calculateChange = (input) => {
-  // Add your code here
+  let remainder = 0;
+  if(input <= 15 && input > 0){
+    const dollers = Math.floor(input);
+    remainder = input - dollers;
+  
+    quarters = Math.floor(remainder / 0.25);
+    remainder -= quarters * 0.25;
+  
+    const dimes = Math.floor(remainder / 0.1);
+    remainder -= dimes * 0.1;
+  
+    const nickles = Math.floor(remainder / 0.05);
+    remainder -= nickles * 0.05;
+  
+    const pennies = Math.round(remainder * 100);
+    
+    if(input === 1){
+      console.log(`${dollers} dollar, ${quarters} quarter, ${dimes} dime, ${nickles} nikel, ${pennies} penny`);
```

Comment:
Nice compact code. It's very readable too, even though it's very succinct. I feel like I wrote a whole novel in comparison.
I did notice, however, that when I run it, the output prints an `undefined` to the console below every correct output. Here's an example of the output when I ran it:
```4 dollars, 2 quarters, 1 dime, 0 nikels, 2 pennies
undefined
9 dollars, 2 quarters, 2 dime, 0 nikels, 4 pennies
undefined
0 quarters, 1 dime, 1 nikels, 1 pennies
undefined
3 quarters, 0 dime, 0 nikels, 0 pennies
undefined
2 dollars, 1 quarters, 1 dime, 1 nikels, 4 pennies
undefined
1 dollar, 0 quarter, 0 dime, 0 nikel, 0 penny
undefined
Error: the number is too large or negative
undefined
```


This is because `calculateChange()` does not actually have a return value, so when it's called inside `console.log()`  on lines 34-48, it returns an `undefined` which then gets printed to the console. 

One possible way to resolve this might be if you remove the `console.log()` statements from within `calculateChange()` and, instead, have `calculateChange()` return a string?